logic/resolvable/Negated.java

```diff
@@ -29,16 +29,21 @@
 import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
 
 public class Negated extends Resolvable<Disjunction> {
-
     // note: we always guarantee unique anonymous IDs within one query
     private final Set<Retrievable> identifiers;
+    private final ResolvableDisjunction disjunction;
 
-    public Negated(Negation negation) {
-        super(negation.disjunction());
+    public Negated(Disjunction disjunction) {
+        super(disjunction);
         this.identifiers = new HashSet<>();
+        this.disjunction = ResolvableDisjunction.of(disjunction);
         pattern().conjunctions().forEach(c -> iterate(c.retrieves()).forEachRemaining(identifiers::add));
     }
 
+    public ResolvableDisjunction disjunction() {
+        return disjunction;
+    }
+
```

Comment:
Uses negated in place of ResolvableNegation.
Breaks the symmetry between the pattern.* and ResolvableFormula.* classes